index.js

```diff
@@ -200,8 +203,18 @@ function adapter(uri, opts){
 
   Redis.prototype.del = function(id, room, fn){
     debug('removing %s from %s', id, room);
+    var hasRoom = this.rooms.hasOwnProperty(room);
     Adapter.prototype.del.call(this, id, room);
-    runCommand('unsubscribe', this.getChannelName(room));
+    if (hasRoom && !this.rooms[room]) {
```

Comment:
Yes, like that.
However, we have to check room is empty before unsubscribe room channel.
And, We have to call Adapter's del() function for manage Adapter's resource.
This implementation is as same as socket.io-redis.
https://github.com/socketio/socket.io-redis/blob/master/index.js#L175-L180